backend/api/views.py

```diff
@@ -0,0 +1,399 @@
+                return render(request,"kiosk.html", details)
+            
+    return HttpResponse("Method Not Allowed")
+
+
+def lang_detect(response1, response2, response3):
+    """
+    Function for detecting the language and its probability
+    ie. [["es:0.9999932356592219"], ["es:0.9999941641326395"], ["es:0.999996766257174"], ["ne:0.5714284946155643", "hi:0.4285697429223616"]]
+    """
+    language_prob = []
+
+    # Make into string elements for data cleaning in all 3 languages
+    detect_en = [str(i) for i in detect_langs(response1)]
+    language_prob.extend(detect_en)
+
+    detect_fr = [str(i) for i in detect_langs(response2)]
+    language_prob.extend(detect_fr)
+
+    detect_es = [str(i) for i in detect_langs(response3)]
+    language_prob.extend(detect_es)
+
+    return list(language_prob)
+
+
+def lang_prob(lang_probability):
+    """
+    Function for seperating the probability and language codes
+    """
+    language_code = []
+
+    # Seperate probs and language codes into 2 lists
+    for i in range(0, len(lang_probability)):
+        code = lang_probability[i][0:2]
+        language_code.append(code)
+        lang_probability[i] = lang_probability[i][3:]
+
+    # Flatten list
+    probs = [float(i) for i in lang_probability]
+
+    return language_code, probs
+
+
+def ISO_639(langauge_code, probability):
+    """
+    Function to get the language ISO
+    """
+    lang_counter = Counter(langauge_code)
+
+    test_val = list(lang_counter.values())[0]
+    res = True
+    
+    for i in lang_counter:
+        if lang_counter[i] != test_val:
+            res = False
+            break
+
+    if res == True:
+        d = defaultdict(list)
+        for key, value in zip(langauge_code, probability):
+            d[key].append(value)
+
+        mean_probs = {key: np.mean(val) for key, val in dict(d).items()}
+
+        ISO_639_2 = str(max(mean_probs, key = mean_probs.get))
+
+    else:
+        ISO_639_2 = max(lang_counter, key = lang_counter.get)
+    
+    return ISO_639_2
+
+
+def exit_input(message):
+    if message.lower() == "exit please":
+        return 1
+    return 0
+
+
+def speak_response(lang_ISO, res_en2lang, lang_voice):
+    """
+    Translate audio response into different languages
+    """
+    if lang_ISO in lang_voice:
+        engine = pyttsx3.init()
+        if lang_ISO == "es":
+            engine.setProperty("rate", 150)  # Speed of speech (words per minute)
+        voices = engine.getProperty("voices")
+        engine.setProperty("voice", voices[lang_voice[lang_ISO]].id)
+        engine.say(res_en2lang)
+        engine.runAndWait()
+
+    # else:
+    #     engine = pyttsx3.init()
+    #     voices = engine.getProperty("voices")
+    #     engine.setProperty("voice", voices[lang_voice[1]].id)
+    #     engine.say("Language not in the system. Please try again in English, French or Spnaish.")
+    #     engine.runAndWait()
```

Comment:
Do we still need this? What if a language that isn't supported is selected, is that possible?

Reply:
We don't need that part of the code. I was testing something and was going to come back to it - however, I will most likely get rid of it in the coming days. 

If the language isn't supported I can make changes to the main function to accommodate for that language. I have made it so that this would be an easy process if required (only a couple of code line changes).